### PR TITLE
Fix nodeproxy release shape

### LINKED ISSUE #3775: NodeProxy crossfade is linear
For audio rate signals, the `NodeProxy` crossfade should be sine shaped (equal power). In fact, it is linear. This is because we use forced release, which (surprisingly) uses a linear shape always.

```supercollider

(
s.waitForBoot {
	Ndef(\x).clear;
	Ndef(\x).play;
	Ndef(\x, { SinOsc.ar(700) * 0.1 });
	Ndef(\x).fadeTime = 2;
	0.5.wait;
	Ndef(\x, { Silent.ar }); Ndef(\x).plot(2.5);
}
)

```

### PR DESCRIPTION
**Pull request by telephon**
Setting the gate of an envelope to a negative value makes a forced release. What remained unnoticed here was that the forced release shape is always linear, regardless of the envelope’s shape. This commit replaces the forced release by a normal release. The fadeTime is set,
so forcing wasn’t necessary.

This fixes #3775.

It changes how crossfading sounds.

testing:
- test how node object replacement and freeing sounds.
- this may change the behaviour of SynthDefs that are separately made
and which *do not use the `fadeTime` parameter to specify the release
time*.

### CODE REVIEW

**Comment by telephon** [comment]
For reference, I'm adding two examples for special cases that are affected by this fix:


```supercollider
// this will still release as expected
SynthDef(\testFadeOut, { Out.ar(\out.kr, EnvGen.kr(Env.asr,
\gate.kr(1), doneAction:2) * PinkNoise.ar * 0.3) }).add;

Ndef(\x, \testFadeOut).play;
Ndef(\x).objects.first.hasFadeTimeControl == false // true
Ndef(\x).fadeTime = 4;
Ndef(\x, { Silent.ar });

// also this will still release as expected
SynthDef(\testFadeOut, { Out.ar(\out.kr, EnvGate.new * PinkNoise.ar *
0.3) }).add;

Ndef(\x, \testFadeOut).play;
Ndef(\x).objects.first.hasFadeTimeControl == true // true
Ndef(\x).fadeTime = 4;
Ndef(\x, { Silent.ar });
```


```
// but not when you "misuse" fadeTime
SynthDef(\testFadeOut, { Out.ar(\out.kr, EnvGen.kr(Env.asr,
\gate.kr(1), doneAction:2) * PinkNoise.ar * 0.3 * SinOsc.ar(\fadeTime * 200 + 100) ) }).add;
```

**Comment by telephon** [comment]
looking for a reviewer …

**Comment by telephon** [comment]
OK, I'll do this. Where would you add it?

**Comment by adcxyz** [comment]
Ndef help file, maybe a subsection:: reserved parameter names
and same in NodeProxy  help file.

File: HelpSource/Classes/NodeProxy.schelp [hunk 1]
```diff
@@ -189,6 +189,23 @@ Ndef(\maus, { LFNoise0.kr(1 ! 8) + 1 }); // now 8 parallel channels of audio are
 
 
 
+subsection::Reserved parameters
+Three parameters are automatically specified if they don't exist in a given UGen function. You can override their use: code::[\out, \gate, \fadeTime]::
```
Comment (LFSaw) [inline HelpSource/Classes/NodeProxy.schelp]: which three parameters?

Comment (brianlheim) [reply HelpSource/Classes/NodeProxy.schelp]: \out, \gate, \fadeTime. this reads OK to me.

Comment (LFSaw) [reply HelpSource/Classes/NodeProxy.schelp]: looks totally ok, I just missed that github code is not wrapping:

screen shot 2018-07-07 at 17 02 59

**Pull request merged**